v2/authorisation/middleware.go

```diff
@@ -175,3 +189,17 @@ func (m PermissionCheckMiddleware) Close(ctx context.Context) error {
 func (m PermissionCheckMiddleware) HealthCheck(ctx context.Context, state *health.CheckState) error {
 	return m.permissionsChecker.HealthCheck(ctx, state)
 }
+
+// GetCollectionIdAttribute provides an implementation of GetAttributesFromRequest. Retrieves and returns
+// header 'Collection-Id' from the request if it exists, otherwise returns an empty map. Never returns an
+// error as the header is not mandatory
+func GetCollectionIdAttribute(req *http.Request) (attributes map[string]string, err error) {
+	attributes = map[string]string{}
+
+	collectionIdAttribute, _ := headers.GetCollectionID(req)
```

Comment:
We should check and handle any potential error that is returned.

Reply:
The only possible returned error here is `ErrHeaderNotFound` which is not an error condition as the `Collection-Id` header is optional so I don't believe any handling is necessary.

Reply:
Could you add a comment above this line explaining that please - it will add clarity for anyone working on this in future.

Reply:
The behaviour is described in the function comment - is this sufficient or is it worth repeating?